packages/core/src/lavadome.mjs

```diff
@@ -31,7 +33,10 @@ export function LavaDome(host, opts) {
     opts = options(opts);
     
     // make exported API tamper-proof
-    defineProperties(this, {text: {value: text}});
+    defineProperties(this, {
+        text: {value: text},
+        copy: {value: copy},
```

Comment:
Re https://github.com/LavaMoat/LavaDome/pull/50/files#r1782176796

Could setting the value of `copy` (and thereby its declaration) be moved to the inner scope of `text`? If so, maybe the need to move `secret` out of scope is no longer here?

Reply:
How would we export `copy` if it's inside `text`?

Reply:
If we assign it inside the handler of `text`, is there a need to?

Reply:
AFAIU, if you move the `copy` function into the `text` function, you closure the secret within the `text` scope instead of the `LavaDome` scope, which doesn't solve the problem, just shifts it elsewhere - the secret still must be scoped somewhere

Reply:
One way to at least prevent one aspect of scope-creep of `secret` itself without compromising on the temper-proofing could be to freeze a property accessor and reassign copy itself instead:

```
// instead of `let secret`
let copy = () => Promise.resolve();

// [...]

  // inside text
  function text(input) {
    copy = async () = {
      // `copy` can now access `input` from `text` directly
    }
  }
}
```

Reply:
sold c8c9cd7

Reply:
I actually just removed c8c9cd7485fb605a97acd5987a0cfd39d7567b4d because:
1. it turns out the way i did it dropped a commit coming from main by accident which could have been a terrible mistake
2. a [younger comment](https://github.com/LavaMoat/LavaDome/pull/50#discussion_r1790955109) made me realize the implementation isn't right, and that also I'm again not convinced this can be solved without storing the secret in some upper scope

For example, I'm not sure how https://github.com/LavaMoat/LavaDome/pull/50#discussion_r1783740760 demonstrates this otherwise, how scoping a function that has access to the secret instead of the secret itself is far better?

A demonstrating PR here could very much help me understand 🙏